controllers/statussync/policy_status_sync.go

```diff
@@ -224,8 +224,12 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 		event := event
 		// sample event.Reason -- reason: 'policy: calamari/policy-grc-rbactest-example'
 		reason := rgx.FindString(event.Reason)
-		if event.InvolvedObject.Kind == policiesv1.Kind && event.InvolvedObject.APIVersion == policiesv1APIVersion &&
-			event.InvolvedObject.Name == instance.GetName() && reason != "" {
+		// Only handle events that match the name and UID of the current Policy
+		if event.InvolvedObject.Kind == policiesv1.Kind &&
```

Comment:
Since a UID is unique, you could avoid all these other conditions to check but it doesn't hurt.

Reply:
> Since a UID is unique, you could avoid all these other conditions to check but it doesn't hurt.

Haha, good point--I was just blindly adding to the list! 😂 
I'll clean it up.